#Notebook 01 :- Data Extraction & Initial Inspection

Objective:-
This notebook covers:
1. Importing required libraries
2. Loading the raw dataset
3. Initial shape and structure inspection
4. Data type verification
5. Missing value overview
6. Basic statistical summary

In [3]:
#imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:

# Load Raw Dataset
df = pd.read_csv("/content/Zomato_Dataset.csv")

print("Dataset loaded successfully!")
print(f"   Rows    : {df.shape[0]}")
print(f"   Columns : {df.shape[1]}")

Dataset loaded successfully!
   Rows    : 45584
   Columns : 20


In [7]:
# Preview first 5 rows of the dataset
df.head()

,ID,Delivery_person_ID,Delivery_person_Age,Delivery_person_Ratings,Restaurant_latitude,Restaurant_longitude,Delivery_location_latitude,Delivery_location_longitude,Order_Date,Time_Orderd,Time_Order_picked,Weather_conditions,Road_traffic_density,Vehicle_condition,Type_of_order,Type_of_vehicle,multiple_deliveries,Festival,City,Time_taken (min)
0,0xcdcd,DEHRES17DEL01,36.0,4.2,30.327968,78.046106,30.397968,78.116106,12-02-2022,21:55,22:10,Fog,Jam,2,Snack,motorcycle,3.0,No,Metropolitian,46
1,0xd987,KOCRES16DEL01,21.0,4.7,10.003064,76.307589,10.043064,76.347589,13-02-2022,14:55,15:05,Stormy,High,1,Meal,motorcycle,1.0,No,Metropolitian,23
2,0x2784,PUNERES13DEL03,23.0,4.7,18.562450,73.916619,18.652450,74.006619,04-03-2022,17:30,17:40,Sandstorms,Medium,1,Drinks,scooter,1.0,No,Metropolitian,21
3,0xc8b6,LUDHRES15DEL02,34.0,4.3,30.899584,75.809346,30.919584,75.829346,13-02-2022,09:20,09:30,Sandstorms,Low,0,Buffet,motorcycle,0.0,No,Metropolitian,20
4,0xdb64,KNPRES14DEL02,24.0,4.7,26.463504,80.372929,26.593504,80.502929,14-02-2022,19:50,20:05,Fog,Jam,1,Snack,scooter,1.0,No,Metropolitian,41


In [8]:
# Check column names and their data types
df.dtypes

,0
ID,object
Delivery_person_ID,object
Delivery_person_Age,float64
Delivery_person_Ratings,float64
Restaurant_latitude,float64
Restaurant_longitude,float64
Delivery_location_latitude,float64
Delivery_location_longitude,float64
Order_Date,object
Time_Orderd,object


In [9]:
# Check missing values in each column
df.isnull().sum()

,0
ID,0
Delivery_person_ID,0
Delivery_person_Age,1854
Delivery_person_Ratings,1908
Restaurant_latitude,0
Restaurant_longitude,0
Delivery_location_latitude,0
Delivery_location_longitude,0
Order_Date,0
Time_Orderd,1731


In [10]:
# Basic statistical summary of numerical columns
df.describe()

,Delivery_person_Age,Delivery_person_Ratings,Restaurant_latitude,Restaurant_longitude,Delivery_location_latitude,Delivery_location_longitude,Vehicle_condition,multiple_deliveries,Time_taken (min)
count,43730.000000,43676.000000,45584.000000,45584.000000,45584.000000,45584.000000,45584.000000,44591.000000,45584.000000
mean,29.566911,4.633774,17.017948,70.229684,17.465480,70.844161,1.023385,0.744635,26.293963
std,5.815064,0.334744,8.185674,22.885575,7.335562,21.120578,0.839055,0.572510,9.384298
min,15.000000,1.000000,-30.905562,-88.366217,0.010000,0.010000,0.000000,0.000000,10.000000
25%,25.000000,4.500000,12.933284,73.170000,12.988453,73.280000,0.000000,0.000000,19.000000
50%,30.000000,4.700000,18.551440,75.897963,18.633934,76.002574,1.000000,1.000000,26.000000
75%,35.000000,4.900000,22.728163,78.044095,22.785049,78.107044,2.000000,1.000000,32.000000
max,50.000000,6.000000,30.914057,88.433452,31.054057,88.563452,3.000000,3.000000,54.000000


In [11]:
# Check unique values in categorical columns
cat_cols = ['Weather_conditions', 'Road_traffic_density', 'Type_of_order',
            'Type_of_vehicle', 'Festival', 'City']

for col in cat_cols:
    print(f"{col} : {df[col].unique()}")
    print()

Weather_conditions : ['Fog' 'Stormy' 'Sandstorms' 'Windy' 'Cloudy' 'Sunny' nan]

Road_traffic_density : ['Jam' 'High' 'Medium' 'Low' nan]

Type_of_order : ['Snack' 'Meal' 'Drinks' 'Buffet']

Type_of_vehicle : ['motorcycle' 'scooter' 'electric_scooter' 'bicycle']

Festival : ['No' 'Yes' nan]

City : ['Metropolitian' 'Urban' 'Semi-Urban' nan]



## Extraction Observations

### Dataset Overview
- Total Rows    : 45,584
- Total Columns : 20

### Missing Values
- Delivery_person_Ratings : 1,908
- Delivery_person_Age     : 1,854
- Time_Orderd             : 1,731
- City                    : 1,200
- multiple_deliveries     : 993
- Weather_conditions      : 616
- Road_traffic_density    : 601
- Festival                : 228

### Data Quality Issues Found
- City column has a typo — 'Metropolitian' should be 'Metropolitan'
- Multiple columns have missing values that need imputation
- Time columns (Time_Orderd, Time_Order_picked) are stored as strings — need conversion

### Next Step
All fixes will be handled in 02_cleaning.ipynb